### ISSUE TITLE
[Bug]: failed to build mo with go 1.20

### ISSUE THREAD

**Issue by xzxiong** [matrixorigin/matrixone#14557]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Branch Name

main

### Commit ID

ad8731be4

### Other Environment Information

```Markdown
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

build with go 1.20 will go error.

<img width="747" alt="企业微信截图_17066835878094" src="https://github.com/matrixorigin/matrixone/assets/3927687/81b3b948-5524-428e-86f7-8b645d3d9f38">

### Steps to Reproduce

```Markdown
1. install go 1.20
2. go clone git@github.com:matrixorigin/matrixone.git
3. exec `cd  matrixone && make`
```

**Comment by xzxiong** [comment]
under pr

**Comment by xzxiong** [comment]
Under review

**Comment by xzxiong** [comment]
![image](https://github.com/matrixorigin/matrixone/assets/3927687/2672a7b2-a65e-49f5-849d-8b63c7a25e2e)

```
Error: pkg/container/types/array_str.go:249:11: SA1019: reflect.StringHeader has been deprecated since Go 1.21 and an alternative has been available since Go 1.20: Use unsafe.String or unsafe.StringData instead. (staticcheck)
	hdr := (*reflect.StringHeader)(unsafe.Pointer(&str))
	         ^
make: *** [Makefile:213: static-check] Error 2
Error: Process completed with exit code 2.
```

**Comment by xzxiong** [comment]
merged
